scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -127,7 +133,195 @@ def setUp(self):
+                res_2=[row_format(r) for r in q.copy(offset=row_count-1)]
+                callback.writeLine('stdout',repr( [len(x) for x in res_1,res_2] +\
+                                                  [
+                                                   (res_1 + res_2 ==
+                                                    [TestCollection+"/"+x for x in ["%04o"%y for y in 0,{max_count}]])
+                                                  ] ))
+                '''),
+            "offset_limit_",
+            lambda: True # lambda : output.replace(" ","").strip() == "[1,1,True]"
+        ), ( #----------------
+            frame_rule('''\
+                q = Query(callback,
+                          columns=["DATA_ID"],
+                          conditions="COLL_NAME = '{{c}}'".format(c=TestCollection))
+                q_iter = iter( q )
+                first_row = next(q_iter)
+                callback.writeLine('stdout',repr([1+len([row for row in q_iter]),
+                                                  q.copy().total_rows()]))
+                '''),
+            "copy_query_",
+            lambda : eval(output) == [ max_count+1 ]*2
+        ), ( #----------------
+            frame_rule('''\
+                from genquery import Option
+                callback.msiModAVUMetadata("-C",TestCollection,"set", "aa", "bb", "cc")
+                callback.msiModAVUMetadata("-C",TestCollection,"add", "aa", "bb", "dd")
+                q = Query(callback,
+                          columns=["META_COLL_ATTR_NAME","COLL_NAME"],
+                          conditions="META_COLL_ATTR_NAME = 'aa' and COLL_NAME = '{{c}}'".format(c=TestCollection)
+                )
+                d0 = [ i for i in q ]
+                d1 = [ i for i in q.copy(options=Option.NO_DISTINCT) ]
+                callback.msiModAVUMetadata("-C",TestCollection,"rmw", "aa", "bb", "%")
+                b = repr([len(d0),len(d1)])
+                callback.writeLine('stdout',b)
+                callback.msiDeleteUnusedAVUs()
+                '''),
+            "no_distinct_",
+            lambda : eval(output) == [1,2]
+        ), ( #----------------
+            frame_rule('''\
+                import itertools
+                q = Query(callback,
+                          columns=["order_asc(DATA_NAME)"],
+                          conditions="DATA_NAME like '03%' and COLL_NAME = '{{c}}'".format(c=TestCollection)
+                )
+                q2 = q.copy(conditions="DATA_NAME like '05%' and COLL_NAME = '{{c}}'".format(c=TestCollection))
+                chained_queries = itertools.chain(q, q2)
+                compare_this = [x for x in ('%04o'%y for y in range({max_count}+1)) if x[:2] in ('03','05')]
+                callback.writeLine('stdout',repr([x for x in chained_queries] == compare_this and len(compare_this)==128))
+                '''),
+            "iter_chained_queries_",
+            lambda : eval(output) is True
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of possibly insecure function - consider using safer ast.literal_eval.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5016258)